src/cycling/BadCyclingPortalImpl.java

```diff
@@ -21,42 +22,119 @@ public int[] getRaceIds() {
 		return new int[] {};
 	}
 
+//We need to implement IllegalNameException for this method
 	@Override
 	public int createRace(String name, String description) throws IllegalNameException, InvalidNameException {
-		// TODO Auto-generated method stub
-		return 0;
+		try 
+		{
+		//Create a new race, an error will be thrown if the name is invalid or illegal
```

Comment:
I dont think you need to comment how to error or why the error is thrown since it should be described in the method